pkg/fuzzer/fuzzer.go

```diff
@@ -0,0 +1,311 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package fuzzer
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/google/syzkaller/pkg/hash"
+	"github.com/google/syzkaller/pkg/ipc"
+	"github.com/google/syzkaller/pkg/rpctype"
+	"github.com/google/syzkaller/prog"
+)
+
+type Fuzzer struct {
```

Comment:
Please add a parallel end-to-end benchmark for pkg/fuzzer with a fast execution function. It will be useful to highlight any serialization bottlenecks and understand  garbage generation profile.
Eventually we will have some fast user-space targets (libsyz) and/or UML/LKL/snapshots. Then when we fuzz with 128/256 cores, synchronization/garbage generation will become a bottleneck. It will be useful to have a solid, efficient core of the fuzzer.